app/controllers/api/integrations_controller.rb

```diff
@@ -6,6 +6,8 @@ class Api::IntegrationsController < Api::BaseController
   before_action :authorize
   before_action :hide_for_apiap, only: :edit
   before_action :find_registry_policies, only: %i[edit update]
+  before_action :instantiate_presenter, only: :update
+  before_action :redirect_to_show, if: :apiap?, only: :edit
```

Comment:
This won't even get to execute while we keep `hide_for_apiap`.
`redirect_to_show` need to be defined too.